core/domain/suggestion_services.py

```diff
@@ -853,3 +874,76 @@ def get_community_contribution_stats():
 
     return create_community_contribution_stats_from_model(
         community_contribution_stats_model)
+
+
+def _update_community_contribution_stats_due_to_suggestions_transactional(
+        suggestions, amount):
+    """Updates the community contribution stats counts associated with the given
+    suggestions by the given amount.
+
+    Args:
+        suggestions: list(Suggestion). Suggestions that may update the counts
+            stored in the community contribution stats. Only suggestion types
+            that are offered on the Contributor Dashboard are accounted for.
+        amount: int. The amount to adjust the counts by.
+    """
+    stats_model = suggestion_models.CommunityContributionStatsModel.get()
+    for suggestion in suggestions:
+        # This method does nothing for suggestion types that are not offered on
+        # the Contributor Dashboard.
+        if suggestion.suggestion_type == (
+                suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT):
+            continue
+        else:
+            stats_model = (
+                suggestion_models.CommunityContributionStatsModel.get())
+            if suggestion.suggestion_type == (
+                    suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+                if suggestion.language_code not in (
+                        stats_model.translation_suggestion_counts_by_lang_code):
+                    stats_model.translation_suggestion_counts_by_lang_code[
+                        suggestion.language_code] = amount
+                else:
+                    stats_model.translation_suggestion_counts_by_lang_code[
+                        suggestion.language_code] += amount
+            elif suggestion.suggestion_type == (
+                    suggestion_models.SUGGESTION_TYPE_ADD_QUESTION):
+                stats_model.question_suggestion_count += amount
+
+        # Create a community contribution stats object to validate the updates.
+        stats = create_community_contribution_stats_from_model(stats_model)
+        stats.validate()
+
+    stats_model.put()
+
+
+def _update_community_contribution_stats_due_to_suggestions(
+        suggestions, amount):
+    """Updates the community contribution stats counts associated with the given
+    suggestions by the given amount. The GET and PUT is done in a single
+    transaction to avoid loss of updates that come in rapid succession.
+
+    Args:
+        suggestions: list(Suggestion). Suggestions that may update the counts
+            stored in the community contribution stats. Only suggestion types
+            that are offered on the Contributor Dashboard are accounted for.
+        amount: int. The amount to adjust the counts by.
+    """
+    transaction_services.run_in_transaction(
+        _update_community_contribution_stats_due_to_suggestions_transactional,
+        suggestions, amount)
+
+
+def _update_community_contribution_stats_due_to_suggestion(suggestion, amount):
```

Comment:
I think it's OK to drop this and just construct a list in the calling functions, for simplicity. It seems reasonable for this to always be a "plural" operation.